### ISSUE TITLE
Why deprecate the browsers option?

### ISSUE THREAD

**Issue by bensampaio** [postcss/autoprefixer#1222]
I was about to upgrade to 9.6.0 when I saw that the `browsers` option was being deprecated. I am wondering why is that and what should then be the solution for my use case.

In the project I am working on I need to generate a bundle for legacy browsers and another one for modern browsers. How can I make the distinction if this configuration has now to be in the package.json?

Additionaly, when I build for development or cypress I choose to build for modern browsers only in order for the build to take less time. How can I make these distinctions if the configuration moves to the package.json?

**Comment by ai** [comment]
You can use Browserslist environments

https://github.com/browserslist/browserslist/blob/master/README.md#environments

We deprecated `browers` because avoiding of using Browserslist creates a lot of issues. Also having single config with target browsers will improve industry.

**Comment by bensampaio** [comment]
I came across the `environments` option before but I don't think it covers my use case. I have a webpack config file that exports 3 configs: legacy bundle, modern bundle and server bundle. The environment could be set to production but each config needs a different browser configuration. How am I supposed to configure browserslist in such a case? And how is this better for the industry?

**Comment by ai** [comment]
You can use environments, not just `production`/`development` (I will change docs).

You can create `legacy`, `modern`, and `server` environments.

Is it what do you looking for?

**Comment by ai** [comment]
I fixed docs ebc3f77

Is that better?

**Comment by ai** [comment]
I fixed docs ebc3f77

Is this better?

**Comment by bensampaio** [comment]
No, this is not what I meant. I have 3 environments (`NODE_ENV`): `development`, `production` and `test`. Creating environments for `modern`, `legacy` and `server` would only make my code more difficult and would still not distinguish between `development` and `production`. I also never have the need to write code just for `legacy` or just for `modern` since babel and autoprefixer take care of transpiling the code to the correct target.

What I meant is that I have 3 webpack configurations inside my `webpack.config.js`. Those configs are executed in one go. The environment is the same for the 3 configs. One config is configured for `legacy`, another for `modern` and another for `server` (these are config [names](https://webpack.js.org/configuration/other-options#name), not environments). If I would create the `legacy`, `modern` and `server` environments, as you suggested, I would need to run 3 separate instances of webpack (at least I cannot think of any other way). In any case, as I mentioned above, these environments would not add any value since I still need the distinction between `development` and `production`.

The solution I can think of is to use the `EnvironmentPlugin` of webpack to set the `BROWSERSLIST_ENV` per webpack configuration. However, I think that the `EnvironmentPlugin` only exposes these env variables to webpack or am I wrong?

In any case, this feels weird to me because now my webpack configs will be completely disconnected from their targets config. This only makes the configuration more difficult to understand for other developers in my team.

**Comment by ai** [comment]
@bensampaio 

> No, this is not what I meant. I have 3 environments (NODE_ENV): development, production and test. Creating environments for modern, legacy and server would only make my code more difficult and would still not distinguish between development and production. 

Browserslist’s environments != `NODE_ENV` (sorry for confusing name). It is just named a list of target browsers. You should use `env` option in Autoprefixer to select environment.

> What I meant is that I have 3 webpack configurations inside my webpack.config.js. Those configs are executed in one go. The environment is the same for the 3 configs.

1. Define environments in Browserslist config.
    ```js
    "browserslist": {
      "modern": [
        "last 2 chrome version",
        "last 2 firefox version",
        "last 2 safari version"
      ],
      "legacy": [
        "defaults"
      ]
    }
    ```
2. Replace `browsers` to `env`:
    ```diff json
    - autoprefixer({ browsers: ['defaults'] })
    + autoprefixer({ env: 'legacy' })
    ```

**Comment by bensampaio** [comment]
Oh that makes a lot more sense. I didn't have time to try it it out but sounds like it should work. Is there a similar option for @babel/preset-env? I need to keep both in sync.

**Comment by ai** [comment]
No. Babel still did not accepted PR with `env` support
https://github.com/babel/babel/pull/9161

You can set `BROWSERSLIST_ENV` environment variable for now.

**Comment by bensampaio** [comment]
I am trying this out now that babel supports browserslist environments too. I managed to make @babel/preset-env work with the `browserslistEnv` option. But autoprefixer says "No browsers selected" with the exact same env value. If I remove the environments from my `.browserslistrc` file then autoprefixer already works, but this is not what I want...

My autoprefixer config: `autoprefixer({ env: 'legacy' })`

My @babel/preset-env config:

```
['@babel/preset-env', {
    browserslistEnv: 'legacy',
    corejs: 3,
    useBuiltIns: 'usage',
}]
```

My browserslistrc file:

```
[legacy]
last 2 versions
not dead

[modern]
last 2 Chrome versions
last 2 ChromeAndroid versions
last 2 Firefox versions
last 2 FirefoxAndroid versions
last 2 Safari versions
last 2 iOS versions
last 2 Edge versions

[node]
node 12
```

This browserslistrc value is at the root of my monorepo while the webpack config is inside a package in the monorepo. I thought this could be the problem so I tried to move the browserslistrc to the application package but that didn't make a difference for `autoprefixer`.

Any ideas on what might be going wrong?
